games-fps/alephone/patches/alephone-20180614.patchset

```diff
@@ -0,0 +1,50 @@
+From 02da082b809730816737753066e56123850bc2b8 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Zolt=C3=A1n=20Mizsei?= <zmizsei@extrowerk.com>
+Date: Wed, 20 Jun 2018 20:24:40 +0200
+Subject: Build fix for Haiku
+
+
+diff --git a/Source_Files/Network/SDL_netx.cpp b/Source_Files/Network/SDL_netx.cpp
+index b3b2360..3dc166d 100644
+--- a/Source_Files/Network/SDL_netx.cpp
++++ b/Source_Files/Network/SDL_netx.cpp
+@@ -35,6 +35,9 @@
+ # if defined(__svr4__)
+ #  define BSD_COMP 1 // This is required to get SIOC* under Solaris
+ # endif
++# if defined(__HAIKU__)
++# include <sys/sockio.h>
++# endif
+ # include <sys/ioctl.h>
+ #endif
+ 
+-- 
+2.16.4
+
+
+From 7b6aef71f776647a5ec16f423148d692869b0ef3 Mon Sep 17 00:00:00 2001
+From: =?UTF-8?q?Zolt=C3=A1n=20Mizsei?= <zmizsei@extrowerk.com>
+Date: Wed, 20 Jun 2018 21:18:11 +0200
+Subject: Settings folder location
+
+
+diff --git a/Source_Files/CSeries/cspaths_sdl.cpp b/Source_Files/CSeries/cspaths_sdl.cpp
+index 6a5d319..d4f5243 100644
+--- a/Source_Files/CSeries/cspaths_sdl.cpp
++++ b/Source_Files/CSeries/cspaths_sdl.cpp
+@@ -174,7 +174,11 @@ static std::string _get_local_data_path()
+ 	{
+ 		const char *home = getenv("HOME");
+ 		if (home)
+-			local_dir = std::string(home) + "/.alephone";
++			#ifndef __HAIKU__
++				local_dir = std::string(home) + "/.alephone";
++			#else
++				local_dir = std::string(home) + "/config/settings/alephone";
```

Comment:
```diff
+#ifndef __HAIKU__
 			local_dir = std::string(home) + "/.alephone";
+#else
+			local_dir = std::string(home) + "/config/settings/alephone";
+#endif
```

Well, this could even be improved with `finddir()`, but I guess this is OK as is.

Reply:
Since this is a C++ program, it should be trivial to add `find_directory()` support, so I would highly recommend it. Feel free to ask for assistance if you are not familiar with `find_directory()` :)